Sources/UnidirectionalBinding.swift

```diff
@@ -73,29 +73,39 @@ public protocol BindingTargetProtocol: class {
 ///
 /// - parameters:
 ///   - target: A target to be bond to.
-///   - signal: A signal to bind.
+///   - source: A source to bind.
 ///
 /// - returns: A disposable that can be used to terminate binding before the
-///            deinitialization of the target or the signal's `completed`
+///            deinitialization of the target or the source's `completed`
 ///            event.
 @discardableResult
 public func <~
 	<Target: BindingTargetProtocol, Source: BindingSourceProtocol>
 	(target: Target, source: Source) -> Disposable?
 	where Source.Value == Target.Value, Source.Error == NoError
 {
-	let disposable = source.observe(Observer(value: { [weak target] in target?.consume($0) }))
+	// Alter the semantics of `BindingTarget` to not require it to be retained.
+	// This is done here--and not in a separate function--so that all variants
+	// of `<~` can get this behavior.
+	let observer: Observer<Source.Value, NoError>
+	if let target = target as? BindingTarget<Source.Value> {
```

Comment:
I am not sure about the behavior. But there seems no point to assume the need of custom binding logic for varying types of sources.

Though I believe if you define it in the protocol, the matching priority would likely be given to the target.